### ISSUE TITLE
My collection box does not show all collections

### ISSUE THREAD

**Issue by jdaKoko** [Japan-Digital-Archives/Japan-Digital-Archive#989]
I have 143 collections under user name Kim, http://jdarchive.org/ja/user/421.  My collection box only shows 114 collections. I cannot open between id913885~913915. Therefore, I cannot add new items to those collections.
If there is a limit to the number of collections each user can create, we should specify that to users.

I already have four user name accounts.  I do not want to create more user accounts.  I would prefer that my collection box shows all the collections created by under each user name.

![my collection box_kim](https://cloud.githubusercontent.com/assets/8831502/8267907/92c029c8-173f-11e5-88a1-6fde91e2ba98.png)

**Comment by horak** [comment]
Looking into this now!

**Comment by horak** [comment]
Koko, it would be great to be able to login in to your account to test this out firsthand and see what errors are thrown, do you think you could change your password and maybe provide the credentials in an email for the account that's giving you trouble?

**Comment by horak** [comment]
@jdaKoko, think you could do the above?

**Comment by horak** [comment]
@jdaKoko, so you're unable to locate collections you made early on, but you are able to access your latest collections correct? Do you have this exact same problem with your other accounts or is this limited to only your 421 account?

**Comment by jdaKoko** [comment]
@horak  Yes, you are correct. my other account "Koko Howell" has the same problem.  http://jdarchive.org/en/user/407
![filtering koko howell collection](https://cloud.githubusercontent.com/assets/8831502/8825146/c7da535e-30b7-11e5-813a-693612123755.png)
The Koko Howell account contains 116 collections, but the first 13 collections (id 885180 ~ 1475720) are not showing up on the collection box.

**Comment by horak** [comment]
I see, ok cool well at least it's consistent, that'll make it easier to solve. @jcuanm, I think the first step here is figuring out if all 143 collections are actually being sent over in the network request!

**Comment by jcuanm** [comment]
I am currently trying to figure out how the collection box works to better see how I can solve the problem. 

As a side note, whenever I login to my account on my dev server, the page shows "none" instead of my username. This is only a problem on my dev server, not on the actual site.

![image](https://cloud.githubusercontent.com/assets/8813674/9422222/5cede614-4840-11e5-8399-f4bf66b3889a.png)

**Comment by jcuanm** [comment]
Furthermore, in my dev server, every time I login and navigate to a new page, I am automatically logged out. The weird thing is that I do not experience this problem on the actual website (even though I have pulled from the upstream and updated my dev server).

**Comment by jcuanm** [comment]
Also, on the actual JDA site, while I try to add new items to my collection box, they are not added. They just float back to their original position on the page. 

Unrelated Question: How does one delete a collection item?

**Comment by jcuanm** [comment]
Actually, I've been playing with the original site and now the drag-and-drop feature for adding items to collections works fine. I don't know why...

**Comment by jcuanm** [comment]
Figured out that the items that are in the explore page run from oldest to newest posts. This also corresponds to the id that the posts have: the larger the id is, the older the post is.

**Comment by jcuanm** [comment]
Found that the collection box stops updating/adding the correct number of items after 100. After 100 posts are added, things start getting weird. The items that are added to the collection box are indeed added to the collection; however the number in the collection box reverts to 97 (100 items if you count the 3 items that are displayed in the collection box) whenever you refresh the page. If you continue adding items to the collection box once things are weird, the collection box jumps to the correct number of items after the numbers 98 or 99 appear on the collection box. This is good since the posts are being added successfully; we just have to fix the number that is displayed.

**Comment by jdaKoko** [comment]
@jcuanm 
Thank you very much for figuring out the problems with the collection box. It's been bothering me for a while. Actually, however, my request is about yet another problem. I have gathered quite a lot of collections on various topics (each with varying number of individual items) under 5 different user names. Two of my user names Kim and Koko Howell have quite a lot of collections. The one I inquired about, Kim, has 143 collections. I have found that, once the number of collections in a given account exceeds either 100 or 114 (it's not clear to me), it becomes impossible to locate the earliest collections gathered under that user name. As you can imagine, this is a huge problem for curation.

The terminology is confusing since the word "collection" is used in various senses. So the "collection box" problem you noticed refers to the set of individual items in a collection, while the problem I'm having is at a higher level, i.e., of groups of collections, each with many individual items.

**Comment by jcuanm** [comment]
@jdaKoko 
I apologize but I'm still a little confused as to which collection box is being referred to. When you refer to your collection box, are you referring to the one highlighted in red here:

![image](https://cloud.githubusercontent.com/assets/8813674/9425671/f8a822bc-48cd-11e5-98fb-0de1c150174a.png)

And the set of individual items is this entire section:

![image](https://cloud.githubusercontent.com/assets/8813674/9425684/8ea0aaaa-48ce-11e5-977d-ec089ed1cee9.png)

**Comment by jdaKoko** [comment]
@jcuanm please read the issue I assigned originally.  If you do not understand, I can meet you in person in September.

**Comment by jdaKoko** [comment]
@jcuanm @meeshmon @horak If someone could fix this problem soon, I would appreciate it very much. I have been waiting a long time for this problem to be solved.

**Comment by horak** [comment]
Should be fixed, let me know if it isn't! @jdaKoko

**Comment by jdaKoko** [comment]
@horak I do not see all of my collections in my Kim account yet. 
The original request was below. Please see the top of #989.

I have 143 collections under user name Kim, http://jdarchive.org/ja/user/421. My collection box only shows 114 collections. I cannot open between id913885~913915. Therefore, I cannot add new items to those collections.
If there is a limit to the number of collections each user can create, we should specify that to users.

I already have four user name accounts. I do not want to create more user accounts. I would prefer that my collection box shows all the collections created by under each user name.

**Comment by horak** [comment]
I may have posted that request a bit too soon, can you try again now?

**Comment by jdaKoko** [comment]
I am sorry, neither the Kim account nor the Koko Howell account shows all collections yet.

**Comment by horak** [comment]
Hmm, that's very strange. I was able to resolve this for my own account. Are you seeing more collections than you did previously?

**Comment by jdaKoko** [comment]
No, I am not seeing more collections than previously. Let me clarify the issues I am having. I have made quite a few collections under two different user names. One problem is that I cannot see all of my collections for either user name (I explain the details below) in either "my collection box" or the "collection field" in the bookmarklet. The other problem is that, once a particular collection has disappeared from "my collection box" and the "collection field" in the bookmarklet, it becomes impossible to update--regardless of whether it has been published or not. There does not seem to be a particular number of collections above which they disappear from "my collection box" and the "collection field" in the bookmarklet (again, see below for specific details). (Let me emphasize that this is a completely separate issue from the fact that there is apparently a limit to the number of items that can go into any individual collection.)

So, did you increase item numbers in one collection?  My request was to see all of my collections in "my collection box" on the JDA site and on the drop down menu in the "collection field" in the bookmarklet when I sign into my accounts. They are not showing all of my creations, with the result that I cannot add new items into ones created earlier.  This issue is very important because I need to be able to update all of my collections as part of my curation activities.  

Under my Kim (id421) account I have 144 collections, but I can only see 112 collections in the "collection field" in the bookmarklet and "my collection box" on the JDA site. 
Under my Koko Howell (id407) account I have 117 collections; I can see 103 only collections.  
The attached images are from "my collection box" and the "collection field" in the bookmarklet.

You have said that you can see everything, which makes me wonder if the problem is not the number of collections per se but rather the age of my accounts. That is, in the past EJ would sometimes make fixes that prevented the problem from occurring in the future without resolving already outstanding instances (for example, he fixed problems with the "scope" and "date" fields in a way that they now work fine when I contribute new items, but I still have to go back and manually fix older items that had already been contributed before he made the fix on January 23, 2013).

![screen shot 2016-01-23 at 10 03 32 am](https://cloud.githubusercontent.com/assets/8831502/12530863/51e9566a-c1ba-11e5-933b-7dddf62e2c7f.png)
![screen shot 2016-01-23 at 10 14 05 am](https://cloud.githubusercontent.com/assets/8831502/12530864/5338a49e-c1ba-11e5-984e-f54c9683adf9.png)

**Comment by horak** [comment]
Thanks for that extended description. I did in fact expand the number of items that display within users' "collection-box"es. I'm going to approach the collection-box issue on the main JDA site first, then i'll get to the bookmarklet (which I'm less familiar with). 

The account age may be the problem, since my account if far newer. Also, there was a strict cutoff with my account, exactly at 100, which was explained in the code I changed. 

I'm going to reach out to you via email to see if I can get access to your account to do some testing.
